### PR TITLE
=pro fix osgi plugin muting AutomaticModuleName

### PR DESCRIPTION
**Pull request by ktoso**
Resolves https://github.com/akka/akka/issues/24520

We should also modularize akka typed right away for 2.5.10

### CODE REVIEW

File: project/AutomaticModuleName.scala [hunk 1]
```diff
@@ -17,7 +18,10 @@ import sbt.Keys._
 object AutomaticModuleName  {
   private val AutomaticModuleName = "Automatic-Module-Name"
 
-  def settings(name: String): Seq[Def.Setting[Task[Seq[PackageOption]]]] = Seq(
-    packageOptions in (Compile, packageBin) += Package.ManifestAttributes(AutomaticModuleName → name)
+  def settings(name: String): Seq[Def.Setting[_]] = Seq(
+    packageOptions in (Compile, packageBin) += Package.ManifestAttributes(AutomaticModuleName → name),
+    // explicitly pass it to the osgi plugin, otherwise it would lose that setting
+    // This should be likely fixed in sbt-osgi, such that it does pick up package options as additional headers
```
Comment (johanandren) [inline project/AutomaticModuleName.scala]: Is there a ticket on that side to refer to?

**Review by johanandren — APPROVED** [review]
LGTM

**Comment by johanandren** [comment]
PLS BUILD

**Review by patriknw — APPROVED** [review]
LGTM, I guess

File: project/AutomaticModuleName.scala (hunk 1 above)
Comment (ktoso) [reply project/AutomaticModuleName.scala]: I'll fix it there now as well and do that, perhaps no need for the fix here then...

**Comment by ktoso** [comment]
FYI, both info included in result:

```
$ ./project/scripts/validate-automatic-module-names
Checking: ../akka-actor/target/akka-actor_2.11-2.5-SNAPSHOT.jar
 Automatic-Module-Name: akka.actor
 Tool: Bnd-2.4.0.201411031534
```

**Comment by ktoso** [comment]
I fixed the plugin instead of our build. please have a look if ok to keep my sanity checking script in there.

**Comment by ktoso** [comment]
Merge + squash at will

**Pull request merged**